app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -181,11 +181,35 @@ def storage_service_resource_attachments
     end
   end
 
-  # This method changes capability name field from 'abstract_capability' to 'name'
-  def parse_capabilities(capabilities, field_name)
+  def parse_ems_capabilities(capabilities)
+    return capabilities unless capabilities.is_a?(Array)
+
+    caps_hash = {}
+    capabilities.each do |capability|
+      name = capability.abstract_capability
+      (caps_hash[name] ||= []) << {"uuid" => capability.uuid, "value" => capability.value}
+    end
+    caps_hash
+  end
+
+  def parse_possible_capabilities(capabilities, field_name)
+    caps = JSON.parse(capabilities)
+    caps_hash = {}
+    caps.each do |capability|
+      name = capability[field_name]
+      (caps_hash[name] ||= []) << capability["value"]
+    end
+    caps_hash
+  end
+
+  def parse_service_capabilities(capabilities, field_name)
     caps = JSON.parse(capabilities)
+    caps_hash = {}
     caps.each do |capability|
-      capability['name'] = capability.delete(field_name)
+      name = capability[field_name]
+      caps_hash[name] = capability["value"]
     end
+    caps_hash
```

Comment:
Similar suggestion using to_h over a local-variable just to set in the loop then return it